sycl/doc/extensions/experimental/sycl_ext_oneapi_bindless_images/ZE_experimental_bindless_images.asciidoc

```diff
@@ -0,0 +1,621 @@
+zeImageDestroy(hImage);
+```
+
+=== USM
+
+```cpp
+// Assumed constructed device and context
+ze_device_handle_t hDevice;
+ze_context_handle_t hContext;
+
+// Assumed allocated command list
+ze_command_list_handle_t hCommandList;
+
+// Assumed image data on host
+std::vector<float> imageDataHost;
+
+// 2D image dimensions
+size_t imageWidth = 1024;
+size_t imageHeight = 1024;
+
+size_t imageRowPitch;
+size_t imageSlicePitch;
+
+// Create a non-USM image memory allocation descriptor
+ze_image_usm_alloc_exp_desc_t imageUSMAllocDesc = {
+  ZE_STRUCTURE_TYPE_IMAGE_USM_ALLOC_DESC,       /* stype */
+  nullptr,                                      /* pNext */
+  imageWidth * sizeof(float),                   /* widthInBytes */
+  imageHeight,                                  /* height */
+  0,                                            /* depth */
+  sizeof(float),                                /* elementSizeBytes */
+  &imageRowPitch,                               /* rowPitch */
+  &imageSlicePitch                              /* slicePitch */
+};
+
+// Allocate USM image memory and return the pointer
+void *imageMemDevice;
+zeMemAllocPitchedExp(hContext, &imageUSMAllocDesc, &imageMemDevice);
+
+// Copy from host to device
+// Our extension allows us to copy to image memory before the image handle
+// itself is created
+zeCommandListAppendImageUSMCopyFromHostExp(hCommandList,
+                                           imageMemDevice,
+                                           imageDataHost.data(),
+                                           nullptr /* pDstRegion */,
+                                           nullptr /* hSignalEvent */,
+                                           0 /* numWaitEvents */,
+                                           nullptr /* phWaitEvents */);
+
+
+// Single-precision float image format with one channel
+ze_image_format_t imageFormat = {
+  ZE_IMAGE_FORMAT_LAYOUT_32,      /* layout */
+  ZE_IMAGE_FORMAT_TYPE_FLOAT,     /* type */
+  ZE_IMAGE_FORMAT_SWIZZLE_R,      /* swizzle x -> R */
+  ZE_IMAGE_FORMAT_SWIZZLE_X,      /* swizzle y -> don't care */
+  ZE_IMAGE_FORMAT_SWIZZLE_X,      /* swizzle z -> don't care */
+  ZE_IMAGE_FORMAT_SWIZZLE_X       /* swizzle w -> don't care */
+}
+
+// 2D image descriptor using previously allocated memory (`imageUSMAllocDesc`)
+ze_image_desc_t imageDesc = {
+    ZE_STRUCTURE_TYPE_IMAGE_DESC, /* stype */
+    &imageUSMAllocDesc,           /* pNext */
+    0,                            /* flags, set to read only */
+    ZE_IMAGE_TYPE_2D,             /* type */
+    imageFormat,                  /* format */
+    1024,                         /* width */
+    1024,                         /* height */
+    0,                            /* depth */
+    0,                            /* arrayLevels */
+    0                             /* miplevels */
+};
+
+
+// Create image from memory allocated above
+ze_image_handle_t hImage;
+zeImageCreate(hContext, hDevice, &imageDesc, &hImage);
```

Comment:
dont understand this. We created imageMemDevice with zeMemAllocPitchedExp, and then initialize it with zeCommandListAppendImageUSMCopyFromHostExp. How, hImage is associated with imageMemDevice when calling zeImageCreate? Should we have a new struct that contains imageMemDevice , and then pass that struct to imageDesc->pNext, so that way zeImageCreate knows it has to associate the hImage handle with the non-USM memory previously created? something like:

```cpp
ze_mem_image_pitched_desc_t pitchedDesc {};
pitchedDesc->ptr = imageMemDevice ;

imageDesc->pNext = &pitchedDesc;

// Create image from memory allocated above
ze_image_handle_t hImage;
zeImageCreate(hContext, hDevice, &imageDesc, &hImage);
```